macro/src/enum_convert.rs

```diff
@@ -0,0 +1,86 @@
+use quote::{quote, ToTokens};
+use syn::parse::{Parse, ParseStream};
+use syn::punctuated::Punctuated;
+use syn::*;
+
+// An attribute that is a list of idents
+pub struct EnumConvertAttribute {
+    path: Path,
+
+    needs_wildcard: bool,
```

Comment:
Question: have we decided yet what to do about when non exhaustive enums in Rust get new fields, but the enums in C++ are exhaustive?

Reply:
C++ doesn't have "exhaustive" enums, no?

Reply:
So should we give the C++ enums version numbers or something? I don't remember if we decided on a policy yet

Reply:
I guess it's less of a problem for enums than structs. As long as the Diplomat-generated enums are `#[repr(u8)]`, we should be fine from a C++ point of view.

Reply:
Why would we have to? Adding variants will not make code stop compiling, like it does in rust

Reply:
Or do we? Maybe it was on `#[derive(ULE)]` where I remember us enforcing repr(u8), not in diplomat.